src/app/clusters/operational-state-server/operational-state-server.cpp

```diff
@@ -60,133 +65,124 @@ CHIP_ERROR OperationalStateServer::Init()
     return CHIP_NO_ERROR;
 }
 
-void OperationalStateServer::Shutdown()
-{
-    InteractionModelEngine::GetInstance()->UnregisterCommandHandler(this);
-}
+// todo change CHIP_ERROR -> Status?
 
-// This function is called by the interaction model engine when a command destined for this instance is received.
-void OperationalStateServer::InvokeCommand(HandlerContext & handlerContext)
+CHIP_ERROR Instance::SetCurrentPhase(const DataModel::Nullable<uint8_t> & aPhase)
 {
-    ChipLogDetail(Zcl, "OperationalState: InvokeCommand");
-    switch (handlerContext.mRequestPath.mCommandId)
-    {
-    case Commands::Pause::Id:
-        ChipLogDetail(Zcl, "OperationalState: Entering handling Pause state");
-
-        HandleCommand<Commands::Pause::DecodableType>(
-            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandlePauseState(ctx, req); });
-        break;
-
-    case Commands::Resume::Id:
-        ChipLogDetail(Zcl, "OperationalState: Entering handling Resume state");
-
-        HandleCommand<Commands::Resume::DecodableType>(
-            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleResumeState(ctx, req); });
-        break;
+    // todo check it the value is valid else return Protocols::InteractionModel::Status::ConstraintError;
 
-    case Commands::Start::Id:
-        ChipLogDetail(Zcl, "OperationalState: Entering handling Start state");
-
-        HandleCommand<Commands::Start::DecodableType>(
-            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleStartState(ctx, req); });
-        break;
-
-    case Commands::Stop::Id:
-        ChipLogDetail(Zcl, "OperationalState: Entering handling Stop state");
-
-        HandleCommand<Commands::Stop::DecodableType>(handlerContext,
-                                                     [this](HandlerContext & ctx, const auto & req) { HandleStopState(ctx, req); });
-        break;
+    DataModel::Nullable<uint8_t> oldPhase = mCurrentPhase;
+    mCurrentPhase                         = aPhase;
+    if (mCurrentPhase != oldPhase)
+    {
+        ConcreteAttributePath path = ConcreteAttributePath(mEndpointId, mClusterId, Attributes::CurrentPhase::Id);
+        MatterReportingAttributeChangeCallback(path);
     }
+    return CHIP_NO_ERROR;
 }
 
-void OperationalStateServer::HandlePauseState(HandlerContext & ctx, const Commands::Pause::DecodableType & req)
+CHIP_ERROR Instance::SetCountdownTime(const DataModel::Nullable<uint32_t> & time)
 {
-    ChipLogDetail(Zcl, "OperationalState: HandlePauseState");
-    Commands::OperationalCommandResponse::Type response;
-    Delegate * delegate = OperationalState::GetOperationalStateDelegate(mEndpointId, mClusterId);
-    GenericOperationalError err(to_underlying(ErrorStateEnum::kNoError));
+    mCountdownTime = time;
+    // We should not report this change.
```

Comment:
You are right. Also, since changes in this attribute do not need to be reported, it is easy for the SDK consumer to implement. And because it is an optional attribute, the Delegate can choose not to store this value at all. I have moved this to the Delegate as a pure virtual [here](https://github.com/project-chip/connectedhomeip/pull/28577/commits/143ed433d08cb6e57f094e33458263fe494df1f0).